src/hotspot/share/oops/klass.hpp

```diff
@@ -161,7 +162,6 @@ class Klass : public Metadata {
 
   // Bitmap and hash code used by hashed secondary supers.
   uintx    _bitmap;
-  uint8_t  _hash_slot;
 
   static uint8_t compute_hash_slot(Symbol* s);
```

Comment:
We don't usually put functions in with nonstatic member declarations.  Since you moved hash_slot, can you move this function to the first private section where hash_insert is?

Where you moved hash_slot looks fine.  Doesn't look like it will have negative cache effects, unless it needs to be on a cache line with _bitmap?

Reply:
Thanks Coleen! Good catch! 
I have checked the cacheline boundaries  with pahole, both are in same cache line after moving, that shouldn't be a concern.
But yes, they are related fields and should be stay together. Also checked the layer after moving both  _bitmap and _hash_slot, it looks good, I'll update the PR:
```
protected:

	jint                       _layout_helper;       /*     8     4 */
	const enum KlassKind       _kind;                /*    12     4 */
	jint                       _modifier_flags;      /*    16     4 */
	juint                      _super_check_offset;  /*    20     4 */
	class Symbol *             _name;                /*    24     8 */
	class Klass *              _secondary_super_cache; /*    32     8 */
	class Array<Klass*> *      _secondary_supers;    /*    40     8 */
	class Klass *              _primary_supers[8];   /*    48    64 */
	/* --- cacheline 1 boundary (64 bytes) was 48 bytes ago --- */
	class OopHandle           _java_mirror;          /*   112     8 */
	class Klass *              _super;               /*   120     8 */
	/* --- cacheline 2 boundary (128 bytes) --- */
	volatile class Klass *     _subklass;            /*   128     8 */
	volatile class Klass *     _next_sibling;        /*   136     8 */
	class Klass *              _next_link;           /*   144     8 */
	class ClassLoaderData *    _class_loader_data;   /*   152     8 */
	int                        _vtable_len;          /*   160     4 */
	class AccessFlags         _access_flags;         /*   164     4 */
	traceid                    _trace_id;            /*   168     8 */
	uintx                      _bitmap;              /*   176     8 */
	uint8_t                    _hash_slot;           /*   184     1 */

	/* XXX 1 byte hole, try to pack */

	s2                         _shared_class_path_index; /*   186     2 */
	u2                         _shared_class_flags;  /*   188     2 */

	/* XXX 2 bytes hole, try to pack */

	/* --- cacheline 3 boundary (192 bytes) --- */
	int                        _archived_mirror_index; /*   192     4 */
```

Reply:
I have updated PR to reflect what we have discussed, thanks!